File: src/main/java/com/bajdcc/OP/syntax/test/TestSyntaxLexer.java
package com.bajdcc.OP.syntax.test;

import com.bajdcc.OP.syntax.lexer.SyntaxLexer;
import com.bajdcc.OP.syntax.token.Token;
import com.bajdcc.OP.syntax.token.TokenType;
import com.bajdcc.util.lexer.error.RegexException;

import java.util.Scanner;

public class TestSyntaxLexer {

	public static void main(String[] args) {
		try {
			Scanner scanner = new Scanner(System.in);
			String str = scanner.nextLine();
			SyntaxLexer lexer = new SyntaxLexer();
			lexer.setContext(str);
			Token token;
			for (; ; ) {
				token = lexer.nextToken();
				if (token.kToken == TokenType.EOF) {
					break;
				}
				System.out.println(token.toString());
			}
			scanner.close();
		} catch (RegexException e) {
			System.err.println(e.getPosition() + "," + e.getMessage());
			e.printStackTrace();
		}
	}
}

File: src/main/java/com/bajdcc/LALR1/syntax/test/TestSyntaxLexer.kt
package com.bajdcc.LALR1.syntax.test

import com.bajdcc.LALR1.syntax.lexer.SyntaxLexer
import com.bajdcc.LALR1.syntax.token.Token
import com.bajdcc.LALR1.syntax.token.TokenType
import com.bajdcc.util.lexer.error.RegexException
import java.util.*

object TestSyntaxLexer {

    @JvmStatic
    fun main(args: Array<String>) {
        try {
            val scanner = Scanner(System.`in`)
            val str = scanner.nextLine()
            val lexer = SyntaxLexer()
            lexer.context = str
            var token: Token?
            while (true) {
                token = lexer.nextToken()
                if (token!!.type === TokenType.EOF) {
                    break
                }
                println(token!!.toString())
            }
            scanner.close()
        } catch (e: RegexException) {
            System.err.println(e.position.toString() + "," + e.message)
            e.printStackTrace()
        }

    }
}
